common/modal.js

```diff
@@ -16,6 +16,11 @@
                 if (rejectCB) {
                     rejectCB(response);
                 } else if (typeof response !== "string") {
+                    // there is no internet connection, so templates couldn't be fetched
+                    // set a flag on the response (error) object
+                    if (response.message.includes("Error: [$compile:tpload]")) {
+                        response.online = $window.navigator.onLine;
```

Comment:
You're only adding this attribute when we cannot fetch the template. So instead of using `online` (which someone might think that it's always going to be true), we should use the inverse of it (`offline`) and set it to true in here.